### PR TITLE
SR-1676 Add GitLab CI terraform-quality-checks pipeline template

### PR DESCRIPTION
**Pull request by WillGibson**
# Description

Adds a GitLab CI pipeline template to run some basic checks on Terraform files in a GitLab CI pipeline.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How this has been tested

This has been tested in three separate GitLab CI pipelines by referencing the file from this branch.

## Checklist

- [x] I have performed a self-review of my code
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: gitlab-ci/terraform-quality-checks/gitlab-ci.yml
Comment (WillGibson) [inline gitlab-ci/terraform-quality-checks/gitlab-ci.yml:1]: The pipeline template is mostly just a lift from one of our GitLab repositories.

File: gitlab-ci/terraform-quality-checks/gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
+variables:
+  HAS_DEV_TERRAFORM: 'true'
```
Comment (WillGibson) [inline gitlab-ci/terraform-quality-checks/gitlab-ci.yml:2]: This variable is new.

File: gitlab-ci/terraform-quality-checks/gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    # Use Folded Style (https://yaml.org/spec/1.2-old/spec.html#id2796251) to stop the colons messing things up.
+    - >
+      if ! grep -q "\"terraform_version\": \"$(cat ../../.terraform-version)\"" terraform.tfstate; then
+        echo 'ERROR: Terraform versions in .terraform-version and terraform.tfstate must match' && exit 1;
+      fi
+    - terraform init
+    - terraform validate
+  rules:
+    - if: $ENV == "dev" && $HAS_DEV_TERRAFORM != "false"
+    - if: $ENV == "prod"
```
Comment (WillGibson) [inline gitlab-ci/terraform-quality-checks/gitlab-ci.yml:22]: These rules are new.

File: gitlab-ci/terraform-quality-checks/gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+prevent_unchecked_dev_files:
+  stage: terraform_quality_checks
+  script:
+    # Use Folded Style (https://yaml.org/spec/1.2-old/spec.html#id2796251) to stop the colons messing things up.
+    - >
+      if [ -d env/dev ]; then
+        echo 'ERROR: Project has env/dev, you must set not HAS_DEV_TERRAFORM to "false"' && exit 1;
+      fi
+  rules:
+    - if: $HAS_DEV_TERRAFORM == "false"
```
Comment (WillGibson) [inline gitlab-ci/terraform-quality-checks/gitlab-ci.yml:67]: This step is new.

**Review by ejayesh — APPROVED** [review]
lgtm.

**Pull request merged**